xaitk_saliency/utils/masking.py

```diff
@@ -1,12 +1,64 @@
-from typing import List
+from typing import List, Tuple
 
 import numpy as np
 import PIL.Image
 
 
+def generate_block_masks(
```

Comment:
@brianhhu @vbhavank There is a different implementation of this in the smqtk mono-repo dev/XAI branch [here](https://github.com/Kitware/SMQTK/blob/dev/XAI/python/smqtk/utils/saliency.py#L112). Is this implementation a valid or more optimal strict replacement for this function

Reply:
I'm not sure about this- it looks like this code was mainly Alina's work? I haven't worked with this utils code before.

Reply:
The linked file *and* this function in `Logit_ImageSaliencyAugmenter` are both attributed to Alina via git-blame

Reply:
I believe the difference between the two implementations is the consideration of image pixels at image boundaries. From what I remember, the linked file's function ignores boundary pixels if it cannot fit a complete sliding window, while Logit_ImageSaliencyAugmenter is the right one that factors in boundary pixels. On a related note, this seems to be a stray function that was refactored [here](https://github.com/Kitware/SMQTK/blob/dev/XAI/python/smqtk/algorithms/saliency/sbsm.py#L173) with additional sanity checks.

Reply:
I'm going to leave this as is for now baring specific need to update. That will probably come with the shift to a streaming API that we've talked about.